share-coordinator/src/main/java/org/apache/kafka/coordinator/share/ShareGroupOffset.java

```diff
@@ -81,42 +98,63 @@ private static PersisterStateBatch toPersisterOffsetsStateBatch(ShareUpdateValue
     }
 
     public static ShareGroupOffset fromRecord(ShareSnapshotValue record) {
-        return new ShareGroupOffset(record.snapshotEpoch(), record.stateEpoch(), record.leaderEpoch(), record.startOffset(), record.stateBatches().stream()
-            .map(ShareGroupOffset::toPersisterOffsetsStateBatch).collect(Collectors.toList()));
+        return new ShareGroupOffset(
+            record.snapshotEpoch(),
+            record.stateEpoch(),
+            record.leaderEpoch(),
+            record.startOffset(),
+            record.stateBatches().stream()
+            .map(ShareGroupOffset::toPersisterOffsetsStateBatch)
+                .toList(),
+            record.createTimestamp(),
+            record.writeTimestamp()
+        );
     }
 
     public static ShareGroupOffset fromRecord(ShareUpdateValue record) {
-        return new ShareGroupOffset(record.snapshotEpoch(), -1, record.leaderEpoch(), record.startOffset(), record.stateBatches().stream()
-            .map(ShareGroupOffset::toPersisterOffsetsStateBatch).collect(Collectors.toList()));
+        return new ShareGroupOffset(
+            record.snapshotEpoch(),
+            -1, record.leaderEpoch(),
```

Comment:
nit: I think the `-1,` probably ought to be by itself on a line to match the rest of the formatting.